client/X11/xf_window.c

```diff
@@ -185,30 +185,27 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 		 */
 		startX = startX + xfc->instance->settings->MonitorLocalShiftX;
 		startY = startY + xfc->instance->settings->MonitorLocalShiftY;
-	}
-
-	xf_ResizeDesktopWindow(xfc, window, width, height);
 
-	/* Set the fullscreen state */
-	xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_STATE, 4,
-				fullscreen ? _NET_WM_STATE_ADD : _NET_WM_STATE_REMOVE,
-				xfc->_NET_WM_STATE_FULLSCREEN, 0, 0);
-
-	/* Only send monitor bounds if they are valid */
-	if ((xfc->fullscreenMonitors.top >= 0) &&
-			(xfc->fullscreenMonitors.bottom >= 0) &&
-			(xfc->fullscreenMonitors.left >= 0) &&
-			(xfc->fullscreenMonitors.right >= 0))
-	{
-		xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_FULLSCREEN_MONITORS, 5,
+		/* Set monitor bounds */
+		if (settings->MonitorCount > 1)
```

Comment:
@bjcollins absolutely and basically you're right. The reason for this change is that some window managers (xfwm4 in particular) show a weird behavior if using _NET_WM_FULLSCREEN_MONITORS property where all IDs (top, bottom, right, left) are the same. On the other hand _NET_WM_FULLSCREEN_MONITORS is *not* required for full screen on a single specific monitor because it is and always was sufficient to move the window to the correct position and set _NET_WM_FULLSCREEN afterwards.